app/src/androidTest/java/com/mapbox/maps/testapp/annotation/UpdateAnnotationWithMultiManagersTest.kt

```diff
@@ -0,0 +1,154 @@
+  private lateinit var polygonAnnotationManager: PolygonAnnotationManager
+  private lateinit var polylineAnnotationManager: PolylineAnnotationManager
+  private lateinit var pointAnnotation: PointAnnotation
+  private lateinit var circleAnnotation: CircleAnnotation
+  private lateinit var polylineAnnotation: PolylineAnnotation
+  private lateinit var polygonAnnotation: PolygonAnnotation
+  private lateinit var handler: Handler
+  private val updateTimes = 100
+  private val updateSteps = 0.1
+  private val points = mutableListOf(
+    mutableListOf(
+      Point.fromLngLat(-3.363937, -10.733102),
+      Point.fromLngLat(1.754703, -19.716317),
+      Point.fromLngLat(-15.747196, -21.085074),
+      Point.fromLngLat(-3.363937, -10.733102)
+    )
+  )
+  private val polylinePoints = mutableListOf(
+    Point.fromLngLat(-4.375974, -2.178992),
+    Point.fromLngLat(-7.639772, -4.107888),
+    Point.fromLngLat(-11.439207, 2.798737),
+  )
+
+  @Test
+  fun testUpdateAnnotation() {
+    rule.scenario.onActivity {
+      handler = Handler(it.mainLooper)
+      it.runOnUiThread {
+        mapboxMap.loadStyleUri(Style.MAPBOX_STREETS) {
+          pointAnnotationManager = mapView.annotations.createPointAnnotationManager(mapView)
+          pointAnnotationManager.textFont = listOf("Open Sans Regular")
+          pointAnnotation = pointAnnotationManager.create(
+            PointAnnotationOptions()
+              .withIconColor(ColorUtils.colorToRgbaString(Color.RED))
+              .withIconImage("car-15")
+              .withPoint(Point.fromLngLat(0.0, 0.0))
+          )
+          circleAnnotationManager = mapView.annotations.createCircleAnnotationManager(mapView)
+          circleAnnotation = circleAnnotationManager.create(
+            CircleAnnotationOptions()
+              .withCircleColor(Color.RED)
+              .withCircleRadius(5.0)
+              .withPoint(Point.fromLngLat(0.0, 0.0))
+          )
+
+          polygonAnnotationManager = mapView.annotations.createPolygonAnnotationManager(mapView)
+          polygonAnnotation = polygonAnnotationManager.create(
+            PolygonAnnotationOptions()
+              .withPoints(points)
+              .withData(JsonPrimitive("Foobar"))
+              .withFillColor(Color.RED)
+          )
+
+          polylineAnnotationManager = mapView.annotations.createPolylineAnnotationManager(mapView)
+          polylineAnnotation = polylineAnnotationManager.create(
+            PolylineAnnotationOptions()
+              .withPoints(polylinePoints)
+              .withLineColor(Color.RED)
+              .withLineWidth(5.0)
+          )
+
+          Thread {
+            for (i in 0..updateTimes) {
+              pointAnnotation.point = Point.fromLngLat(
+                pointAnnotation.point.longitude() + updateSteps,
+                pointAnnotation.point.latitude() + updateSteps
+              )
+              pointAnnotationManager.update(pointAnnotation)
+              Thread.sleep(10)
```

Comment:
Why is sleep needed?

Reply:
To simulate the annotation updating of the drag gesture.